cmd/falco/runner.go

```diff
@@ -273,11 +269,80 @@ func (r *Runner) run(v *VCL, mode *RunMode) ([]*plugin.VCL, error) {
 			if !ok {
 				continue
 			}
-			r.printLinterError(lx, le)
+			r.printLinterError(r.lexers[v.Name], le)
 		}
 	}
 
-	return vcls, nil
+	return &plugin.VCL{
+		File: v.Name,
+		AST: vcl,
+	}, nil
+}
+
+func (r *Runner) resolveStatements(statements []ast.Statement) ([]ast.Statement, error) {
+	var resolved []ast.Statement
+
+	for _, stmt := range statements {
+		switch t := stmt.(type) {
+		case *ast.IncludeStatement:
+			module, err := r.resolver.Resolve(t.Module.Value)
+			if err != nil {
+				return nil, err
+			}
+
+			vcl, err := r.parseVCL(module.Name, module.Data)
+			if err != nil {
+				return nil, err
+			}
+
+			vcl.Statements, err = r.resolveStatements(vcl.Statements)
+			if err != nil {
+				return nil, err
+			}
+			resolved = append(resolved, vcl.Statements...)
+		case *ast.IfStatement:
+			if err := r.resolveIfStatement(t); err != nil {
+				return nil, err
+			}
+			resolved = append(resolved, t)
```

Comment:
`If` statement may have a block statement so need to resolve recursively.

Reply:
(also we should probably move this to the code as a comment)

Reply:
The block statement may have `include` statement of embedding [dynamic snippets](https://docs.fastly.com/en/guides/using-dynamic-vcl-snippets#including-dynamic-snippets-in-custom-vcl) inside (and probably include file is also valid).

For example:

```vcl
sub vcl_fetch {
  if (req.restarts > 0) {
    include "snippet::<snippet_name>"; # Embed dynamic snippet here
  }
  ...
}
```

According to [include statement](https://developer.fastly.com/reference/vcl/statements/include/), it is available in all subroutines so the above syntax should be valid and resolve inclusion recursively.